todo-app/src/Components/AddTodoItem.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from "react";
+import { observer, inject } from "mobx-react";
+
+import Button from "./Button";
+import InputField from "./InputField";
+import "./styles/addTodoItem.css";
+
+@inject("todoStore")
+@observer
+export default class AddTodoItem extends Component {
+  state = {
+    active: false,
+    toBeAddedItem: "",
+    toBeAddedDeadline: ""
+  };
+
+  toggleAddForm = event => {
+    this.setState({
+      active: !this.state.active
+    });
+
+    this.state.active && this.clearForm();
+  };
+
+  handleChangeTodoItem = newText => {
+    this.setState({ toBeAddedItem: newText });
+  };
+
+  handleChangeDeadline = newDeadline => {
+    this.setState({ toBeAddedDeadline: newDeadline });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+  };
+
+  clearForm = () => {
+    this.setState({
+      toBeAddedItem: "",
+      toBeAddedDeadline: ""
+    });
+  };
+
+  handleAddTodo = () => {
+    const { toBeAddedItem, toBeAddedDeadline } = this.state;
+    this.props.todoStore.addTodoItem(toBeAddedItem, toBeAddedDeadline);
+    this.setState({ active: false });
+    this.clearForm();
+  }; // good
+
+  showForm = () => {
+    return (
+      <div className="addTodoForm">
+        <form onSubmit={this.handleSubmit}>
+          <div className="formInputWrapper">
+            <InputField
+              type="text"
+              placeholder="Todo item"
+              className="todoInput"
+              value={this.state.toBeAddedItem}
+              handleChange={this.handleChangeTodoItem}
+            />
+            <InputField
```

Comment:
It would be nice to have a datepicker here instead of a text input. At the moment it's very easy to type an invalid date in here and break the app (no error feedback, by the way). There is a [native HTML5 date input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date) (although it's famous for being terrible). There are also many, many third-party datepicker components for React.